core/src/rpc_subscriptions.rs

```diff
@@ -291,6 +301,26 @@ impl RpcSubscriptions {
             self.check_signature(signature, current_slot, bank_forks);
         }
     }
+
+    pub fn add_slot_subscription(&self, sub_id: &SubscriptionId, sink: &Sink<SlotInfo>) {
+        let mut subscriptions = self.slot_subscriptions.write().unwrap();
+        subscriptions.insert(sub_id.clone(), sink.clone());
+    }
+
+    pub fn remove_slot_subscription(&self, id: &SubscriptionId) -> bool {
+        let mut subscriptions = self.slot_subscriptions.write().unwrap();
+        subscriptions.remove(id).is_some()
+    }
+
+    pub fn notify_slot(&self, slot: Slot, parent: Slot, root: Slot) {
+        info!("notify_slot!! {} from {} (root={})", slot, parent, root);
```

Comment:
@sunnygleason - here's the log to look for when trying out this PR.  It contains the basic data that once surfaced through pubsub can be used to built out the live fork graph.

We can prune forks from the graph whenever the `root` slot moves beyond the latest `slot` for that fork